apps/teams/src/app/setup/route.ts

```diff
@@ -0,0 +1,34 @@
+import { RedirectType, redirect } from 'next/navigation';
+import type { NextRequest } from 'next/server';
+import { env } from '@/env';
+import { getTokens } from '@/connectors/auth';
+import { setupOrganisation } from './service';
+
+export async function GET(request: NextRequest): Promise<void> {
+  let redirectUrl = '';
+  try {
+    const { code, organisationId } = validateRequest(request);
+    const tokens = await getTokens(code);
+    await setupOrganisation(organisationId, tokens.accessToken, tokens.refreshToken);
+    redirectUrl = `${env.ELBA_REDIRECT_URL}?source=${env.ELBA_SOURCE_ID}&success=true`;
+  } catch (error: unknown) {
+    redirectUrl = `${env.ELBA_REDIRECT_URL}?source=${env.ELBA_SOURCE_ID}&error=true`;
```

Comment:
Sorry, we did not document this and the template might be out of date, but we are now supporting error code on elba.

```suggestion
   if (error instanceof TeamsError && error.response.status === 401) {
     redirectUrl = `${env.ELBA_REDIRECT_URL}?source=${env.ELBA_SOURCE_ID}&error=unauthorized`;
   } else {
     redirectUrl = `${env.ELBA_REDIRECT_URL}?source=${env.ELBA_SOURCE_ID}&error=internal_error`;
   }
```